tools/cldr-code/src/main/java/org/unicode/cldr/test/CLDRTest.java

```diff
@@ -509,22 +514,17 @@ public void TestDisplayNameCollisions() {
 
             for (Iterator<String> it2 = item.iterator(); it2.hasNext(); ) {
                 String xpath = it2.next();
-                int nameType = CLDRFile.getNameType(xpath);
-                if (nameType < 0) continue;
+                NameType nameType = NameType.fromPath(xpath);
+                if (nameType == NameType.NONE) continue;
                 String value = item.getStringValue(xpath);
-                String xpath2 = maps[nameType].get(value);
+                int nameTypeIndex = nameTypeIntegerMap.get(nameType);
+                String xpath2 = maps[nameTypeIndex].get(value);
                 if (xpath2 == null) {
-                    maps[nameType].put(value, xpath);
+                    maps[nameTypeIndex].put(value, xpath);
                     continue;
                 }
-                collisions.add(
-                        CLDRFile.getNameTypeName(nameType)
-                                + "\t"
-                                + value
-                                + "\t"
-                                + xpath
-                                + "\t"
-                                + xpath2);
+                String theName = nameType.getNameTypeName();
```

Comment:
What is it the name of? There are too many variables like "noun2" -- and reading this code without having written it, I don't know what "theName" really is.

Reply:
Join the party :-) Doing this refactoring, I've also been diving into a lot of unfamiliar and perplexing code. The same variable or parameter may be called a name, or a type, or a code, ... This method in particular is looping through all the paths, so depending on the path, theName could be the name of a language, a script, a territory, ... -- any of the values of the new NameType enum. A path for a language name is associated with NameType.LANGUAGE; a path for a script name is associated with NameType.SCRIPT; etc.